tests/ui/specs/VirtualList/VirtualGridList/VirtualGridListPage.js

```diff
@@ -133,10 +135,52 @@ class VirtualGridListPage extends Page {
 		}, scrollableSelector);
 	}
 
-	itemOffsetTopById (id) {
-		return browser.execute(function (_element) {
-			return _element.getBoundingClientRect().top;
-		}, this.item(id).value);
+	itemOffsetTopById (index) {
+		return browser.execute(function (_listItemSelector, _scrollableSelector, _index) {
+			const itemContent = document.querySelectorAll(_listItemSelector)[_index];
+			const scroller = document.querySelector(_scrollableSelector);
+			return Math.round(scroller.getBoundingClientRect().height + scroller.getBoundingClientRect().top - itemContent.getBoundingClientRect().top);
+		}, listItemSelector, scrollableSelector, index);
+	}
+
+	itemOffsetBottomById (index) {
+		return browser.execute(function (_listItemSelector, _scrollableSelector, _index) {
+			const itemContent = document.querySelectorAll(_listItemSelector)[_index];
+			const scroller = document.querySelector(_scrollableSelector);
+			return Math.round(itemContent.getBoundingClientRect().bottom - scroller.getBoundingClientRect().top);
+		}, listItemSelector, scrollableSelector, index);
+	}
+
+	getItemSize (index = 0) {
+		return browser.execute(function (_listItemSelector, _index) {
+			const itemContent = document.querySelectorAll(_listItemSelector)[_index];
+			const {height: itemHeight, width: itemWidth}  = itemContent.getBoundingClientRect();
+			return {
+				height: itemHeight,
+				width: itemWidth
+			};
```

Comment:
There is double space.

And. Sorry, I just realized you don't need to rename.
```suggestion
			const {height, width} = itemContent.getBoundingClientRect();
			return {
				height,
				width
			};
```
Or How about returning all rect value.
```suggestion
			return itemContent.getBoundingClientRect();
```

Reply:
Suggested change.